numerrin_wrapper/numerrin_mesh.py

```diff
@@ -0,0 +1,448 @@
+from simphony.core.cuba import CUBA
+
+from .numerrin_templates import numname, numvariables
+
+import numerrin
+
+
+class NumerrinMesh(ABCMesh):
+    """ Proxy class to communicate with Numerrin pool mesh data
+
+    Parameters
+    ----------
+    name : str
+        name of mesh
+
+    mesh : ABCMesh
+       mesh to store
+
+    pool : NumerrinPool
+       Numerrin variable pool
+
+    Attributes
+    ----------
+    name : str
+        name of mesh
+
+    mesh : ABCMesh
+       mesh to store
+
+    pool : NumerrinPool
+       Numerrin variable pool
+
+    _uuidToNumLabel : dictionary
+        Mapping from uuid to Numerrin label number
+    _numCellLabelToUuid : dictionary
+        Mapping from Numerrin cell label number to uuid
+    _numFaceLabelToUuid : dictionary
+        Mapping from Numerrin face label number to uuid
+    _numEdgeLabelToUuid : dictionary
+        Mapping from Numerrin edge label number to uuid
+    _numPointLabelToUuid : dictionary
+        Mapping from Numerrin point label number to uuid
+
+
+    """
+
+    def __init__(self, name, mesh, pool):
+        super(NumerrinMesh, self).__init__()
+        self.name = name
+        self.pool = pool
+        self._time = str(0)
+        maps = self.pool.import_mesh(name, mesh)
+        self._uuidToNumLabel = maps[0]
+        self._numPointLabelToUuid = maps[1]
+        self._numEdgeLabelToUuid = maps[2]
+        self._numFaceLabelToUuid = maps[3]
+        self._numCellLabelToUuid = maps[4]
+        # point data
+        for point in mesh.iter_points():
+            self.update_point(point)
+
+    def get_point(self, uuid):
+        """Returns a point with a given uuid.
+
+        Returns the point stored in the mesh
+        identified by uuid. If such point do not
+        exists an exception is raised.
+
+        Parameters
+        ----------
+        uuid
+            uuid of the desired point.
+
+        Returns
+        -------
+        Point
+            Mesh point identified by uuid
+
+        Raises
+        ------
+        Exception
+            If the point identified by uuid was not found
+
+        """
+
+        try:
+            coords = numerrin.getnode(self.pool.ph, self.name,
+                                      self._uuidToNumLabel[uuid])
+            point = Point(coords, uuid)
+            for dkey in numvariables:
+                dataName = numname[dkey]
+                mDataName = self.name + dataName
+                try:
+                    if numerrin.gettype(self.pool.ph, mDataName) == "Function":
+                        dataV = numerrin.getrealfunction(self.pool.ph,
+                                                         mDataName)
+                    else:
+                        dataV = numerrin.getvariable(self.pool.ph,
+                                                     mDataName)
+                    point.data[dkey] = dataV[self._uuidToNumLabel[uuid]]
+                except:
+                    pass
```

Comment:
I still think ignoring all exceptions is not a good idea (e.g. also those possibly arising from ``point.data[dkey]``)

The three numerrin functions ``gettype``, ``getrealfunction``, and ``getvariable`` appear to invoke only ``PyExc_RuntimeError``. Does this translate into the standard ``RuntimeError`` on the Python side?

If yes, it would be better to catch that specific exception, and perhaps take some appropriate actions
(e.g. just to print that getting a valid variable failed unexpectedly).